frontend/src/app.tsx

```diff
@@ -3,14 +3,20 @@ import { BrowserRouter } from 'react-router-dom';
 
 import { AppRouter } from './router';
 import { AuthDataServiceContextProvider } from './services/auth-data-service';
+import { AuthHttpServiceContextProvider } from './services/auth.http-service';
 import { HttpServiceContextProvider } from './services/http-service';
+import { ServiceDocsHttpServiceContextProvider } from './services/service-docs.http-service';
 
 export const App: React.FC = () => {
   return (
     <BrowserRouter>
       <AuthDataServiceContextProvider>
         <HttpServiceContextProvider>
-          <AppRouter />
+          <AuthHttpServiceContextProvider>
+            <ServiceDocsHttpServiceContextProvider>
+              <AppRouter />
+            </ServiceDocsHttpServiceContextProvider>
+          </AuthHttpServiceContextProvider>
```

Comment:
In the long run, this will become super nested :D How about creating a "HttpServiceAndRelatedStuffProvider" (with a better name of course) that lives in the "services" directory (or even better: in the "services/http" directory - see my other comment) and provides all of the services?

```tsx
interface Props {
  children: React.ReactNode;
}
export const HttpServiceAndRelatedStuffProvider: React.FC<Props> = props => {
  return (
    <HttpServiceContextProvider>
      <AuthHttpServiceContextProvider>
         <ServiceDocsHttpServiceContextProvider>
            {props.children}
        </ServiceDocsHttpServiceContextProvider>
      </AuthHttpServiceContextProvider>
    </HttpServiceContextProvider>
  );
}
```